util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java

```diff
@@ -172,70 +233,121 @@ private static ConnectivityState aggregateState(
+    public ChildLbState(Object key, LoadBalancerProvider policyProvider, Object childConfig,
+        SubchannelPicker initialPicker) {
       this.key = key;
       this.policyProvider = policyProvider;
       lb = new GracefulSwitchLoadBalancer(new ChildLbStateHelper());
       lb.switchTo(policyProvider);
       currentPicker = initialPicker;
+      config = childConfig;
+    }
+
+
+    @Override
+    public String toString() {
+      return "Address = " + key
+          + ", state = " + currentState
+          + ", picker type: " + currentPicker.getClass()
+          + ", lb: " + lb.delegate().getClass()
+          + (deactivated ? ", deactivated" : "");
+    }
+
+    public Object getKey() {
+      return key;
+    }
+
+    Object getConfig() {
+      return config;
+    }
+
+    public LoadBalancerProvider getPolicyProvider() {
+      return policyProvider;
+    }
+
+    protected Subchannel getSubchannels(PickSubchannelArgs args) {
+      return getCurrentPicker().pickSubchannel(args).getSubchannel();
+    }
+
+    ConnectivityState getCurrentState() {
+      return currentState;
+    }
+
+    public SubchannelPicker getCurrentPicker() {
+      return currentPicker;
+    }
+
+    public boolean isDeactivated() {
+      return deactivated;
+    }
+
+    @VisibleForTesting
+    LoadBalancer getLb() {
+      return this.lb;
     }
 
-    void deactivate() {
+    protected void setDeactivated() {
+      deactivated = true;
+    }
+
+    protected void deactivate() {
       if (deactivated) {
         return;
       }
 
-      class DeletionTask implements Runnable {
-        @Override
-        public void run() {
-          shutdown();
-          childLbStates.remove(key);
-        }
-      }
-
-      deletionTimer =
-          syncContext.schedule(
-              new DeletionTask(),
-              DELAYED_CHILD_DELETION_TIME_MINUTES,
-              TimeUnit.MINUTES,
-              timeService);
+      shutdown();
+      childLbStates.remove(key);
       deactivated = true;
       logger.log(Level.FINE, "Child balancer {0} deactivated", key);
     }
 
-    void reactivate(LoadBalancerProvider policyProvider) {
-      if (deletionTimer != null && deletionTimer.isPending()) {
-        deletionTimer.cancel();
-        deactivated = false;
-        logger.log(Level.FINE, "Child balancer {0} reactivated", key);
-      }
+    protected void reactivate(LoadBalancerProvider policyProvider) {
       if (!this.policyProvider.getPolicyName().equals(policyProvider.getPolicyName())) {
         Object[] objects = {
             key, this.policyProvider.getPolicyName(),policyProvider.getPolicyName()};
         logger.log(Level.FINE, "Child balancer {0} switching policy from {1} to {2}", objects);
         lb.switchTo(policyProvider);
         this.policyProvider = policyProvider;
+      } else {
+        logger.log(Level.FINE, "Child balancer {0} reactivated", key);
```

Comment:
Whether we need to switch to a different LB provider or can trivially reuse the same one, we want to reset the flag so that the system thinks it is active ("doesn't think it is deactivated" is too many negatives :) )